### PR TITLE
fix: Update examples to show integration and usage of new IRSA submodule

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Update examples to show integration and usage of new IRSA role submodules
- Correct one instance of hard coded AWS DNS suffix to use AWS partition data source

## Motivation and Context
- Better inform users how to use new IRSA module that integrates with this module

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -234,10 +234,6 @@ resource "aws_eks_addon" "this" {
     ]
   }
 
-  # Note: if an addon needs to be provisioned ahead of a node group users will
```
Comment (bryantbiggs) [inline main.tf:237]: No longer need this note - this will most likely stay as its defined

File: examples/eks_managed_node_group/main.tf [hunk 2]
```diff
@@ -421,6 +424,24 @@ module "vpc" {
...
+module "vpc_cni_irsa" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 4.12"
+
+  role_name_prefix      = "VPC-CNI-IRSA"
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv6   = true
+
+  oidc_providers = {
+    ex = {
```
Comment (antonbabenko) [inline examples/eks_managed_node_group/main.tf:436]: `ex` is what you called `main` in another module (`iam`), right? `main` seems more descriptive to me.

Comment (bryantbiggs) [reply examples/eks_managed_node_group/main.tf]: fixed all references in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1882/commits/94c0309a7fd35174ab77462bf509e03fc60a27b3

**Comment by antonbabenko** [comment]
This PR is included in version 18.7.2 :tada:

**Pull request merged**